### PR TITLE
prevent rerender loop in screenshot

### PR DESCRIPTION
**Pull request by hypervillain**
Fixes https://linear.app/prismic/issue/SM-983/[performance]-screenshot-page-renders-continuously-causing-high-cpu

### CODE REVIEW

File: packages/slice-machine/src/hooks/useEditorContent.ts [hunk 1]
```diff
@@ -15,12 +16,12 @@ function useEditorContentOnce({
     slice.mock?.find((m) => m.variation === variationID) ||
     defaultSharedSliceContent(variationID);
 
-  const apiContent = {
+  const apiContent = useMemo(() => ({
     ...(renderSliceMock(Slices.fromSM(slice.model), editorContent) as object),
     id: slice.model.id,
-  };
+  }), []);
```
Comment (mathieud) [inline packages/slice-machine/src/hooks/useEditorContent.ts]: Does adding the dependencies here trigger the bug?

Comment (hypervillain) [reply packages/slice-machine/src/hooks/useEditorContent.ts]: you're right! Updated

**Review by mathieud — APPROVED** [review]

File: packages/slice-machine/pages/[lib]/[sliceName]/[variation]/screenshot.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ const Screenshot: ComponentWithSliceProps = ({ slice, variation }) => {
     variationID: variation.id,
   });
 
-  if (!simulatorUrl) {
+  if (simulatorUrl === undefined) {
```
Comment (MarcMcIntosh) [inline packages/slice-machine/pages/[lib]/[sliceName]/[variation]/screenshot.tsx:32]: can simulator url be an other falsely value?

**Review by MarcMcIntosh — APPROVED** [review]
Just one question :)

**Review by MarcMcIntosh — CHANGES_REQUESTED** [review]
Which branch should this one target?

File: packages/slice-machine/pages/[lib]/[sliceName]/[variation]/screenshot.tsx (hunk 2 above)
Comment (hypervillain) [reply packages/slice-machine/pages/[lib]/[sliceName]/[variation]/screenshot.tsx:32]: no but Typescript complains if you use a non-boolean value in conditions

**Pull request merged**